cfpb/cfgov-refresh
# Article page type

Adds a structured page type to feature multiple related Ask CFPB questions in article format.

## Additions

- Article page type
- Article template

## Notes
The article page type is intended to allow testing of a new way to aggregate Ask CFPB questions. Articles will be associated with one of 4 of the portal categories ("key terms" is not included) and will use a very structured format. The page components include:

### Heading
The heading fields are required and include a title, intro, and portal category.

| Input fields | Page output |
| -- | -- |
|  <img width="954" alt="Screen Shot 2019-05-16 at 12 47 19 PM" src="https://user-images.githubusercontent.com/778171/57882548-ce579b80-77d8-11e9-900b-a53fda79c342.png"> | <img width="645" alt="Screen Shot 2019-05-16 at 12 47 11 PM" src="https://user-images.githubusercontent.com/778171/57882534-c5ff6080-77d8-11e9-8ba7-decdb7c59d9e.png"> |


### Inset links (optional)
Provides a place to feature 2 links to key terms, PDFs, or other related content.

| Input fields | Page output |
| -- | -- |
| <img width="1032" alt="inset links" src="https://user-images.githubusercontent.com/778171/57882202-06121380-77d8-11e9-8733-96133d14a471.png"> | <img width="337" alt="inset-links-output" src="https://user-images.githubusercontent.com/778171/57882211-0d392180-77d8-11e9-81cc-4278fbb1dc2c.png"> |

### Sections & subsections
These output a possible two levels of headings, summaries, and links to answer pages. Summaries are optional at both levels and links are optional at the section level. Anchor links to the top-level sections are also featured in a table of contents at the top of the page.

| Input | Table of contents output | Content output |
| -- | -- | -- |
|  <img width="1238" alt="Screen Shot 2019-05-16 at 12 51 25 PM" src="https://user-images.githubusercontent.com/778171/57883815-2643d180-77dc-11e9-95b4-f4b3da85e986.png"> | <img width="456" alt="Screen Shot 2019-05-16 at 1 16 21 PM" src="https://user-images.githubusercontent.com/778171/57884136-d3b6e500-77dc-11e9-830f-6625d3c8a308.png">  | <img width="735" alt="Screen Shot 2019-05-16 at 1 16 13 PM" src="https://user-images.githubusercontent.com/778171/57884146-d9acc600-77dc-11e9-910f-3e5762783d80.png">  |

## Screenshots
![page](https://user-images.githubusercontent.com/778171/57893737-e808db00-77f8-11e9-9172-795c4a486db7.png)


## Testing

1. Check out branch

2. Run `python cfgov/manage.py migrate`

3. Add an article page and test its fields (an example page built without the template is on build at `consumer-tools/debt-collection/respond-debt-collector/`)
